src/eShop.AppHost/Program.cs

```diff
@@ -61,24 +59,63 @@
     .WithReference(webHooksApi)
     .WithEnvironment("IdentityUrl", idpHttps);
 
-var webApp = builder.AddProject<Projects.WebApp>("webapp")
+var webApp = builder.AddProject<Projects.WebApp>("webapp", "https")
+    .WithExternalHttpEndpoints()
     .WithReference(basketApi)
     .WithReference(catalogApi)
     .WithReference(orderingApi)
     .WithReference(rabbitMq)
-    .WithReference(openAi, optional: true)
-    .WithEnvironment("IdentityUrl", idpHttps)
-    .WithLaunchProfile("http");
+    .WithEnvironment("IdentityUrl", idpHttps);
+
+// set to true if you want to use OpenAI
+bool useOpenAI = false;
+if (useOpenAI)
+{
+    const string openAIName = "openai";
+    const string textEmbeddingName = "text-embedding-ada-002";
+    const string chatModelName = "gpt-35-turbo-16k";
+
+    // to use an existing OpenAI resource, add the following to the AppHost user secrets:
+    // "ConnectionStrings": {
+    //   "openai": "Key=<API Key>" (to use https://api.openai.com/)
+    //     -or-
+    //   "openai": "Endpoint=https://<name>.openai.azure.com/" (to use Azure OpenAI)
+    // }
+    IResourceBuilder<IResourceWithConnectionString> openAI;
+    if (builder.Configuration.GetConnectionString(openAIName) is not null)
+    {
+        openAI = builder.AddConnectionString(openAIName);
+    }
+    else
+    {
+        // to use Azure provisioning, add the following to the AppHost user secrets:
+        // "Azure": {
+        //   "SubscriptionId": "<your subscription ID>"
+        //   "Location": "<location>"
+        // }
+        openAI = builder.AddAzureOpenAI(openAIName)
```

Comment:
There is an issue in P5 to be aware of where you can't send two OpenAI deployments to an OpenAI Account at once. You need to either batch them or set a dependency. We've @sebastienros fixed it for P6 but it means that you may need to trigger a deployment twice as a work around.

Reply:
You can F5 twice and it will work.  Or you can configure to talk to an existing service.

We can wait for P6 for the real fix.